common/djangoapps/student/models.py

```diff
@@ -1605,23 +1605,31 @@ def emit_event(self, event_name, enterprise_uuid=None, request=None):  # pylint:
                 extra_segment_properties = {
                     'studio_request': studio_request
                 }
+                exception_raised = False
                 if not studio_request and should_send_redesign_email():
                     if not request:
                         request = crum.get_current_request()
 
                     marketing_root_url = settings.MKTG_URLS.get('ROOT')
-                    course_dates_list = get_course_dates_for_email(self.user, self.course.id, request)
                     course_run_fields = [
                         'key', 'title', 'short_description', 'marketing_url', 'pacing_type', 'min_effort',
                         'max_effort', 'weeks_to_complete', 'enrollment_count', 'image', 'staff',
                     ]
-                    owners, course_run = None, None
+                    owners, course_run, course_dates_list = None, None, []
                     try:
+                        course_dates_list = get_course_dates_for_email(self.user, self.course.id, request)
                         course_uuid = get_course_uuid_for_course(str(self.course_id))
                         owners = get_owners_for_course(course_uuid=course_uuid)
                         course_run = get_course_run_details(str(self.course_id), course_run_fields)
                     except Exception:   # pylint: disable=broad-except
-                        pass
+                        exception_raised = True
+                        if event_name and self.course_id:
```

Comment:
Is this condition necessary? In which case these can be null?

Reply:
Not necessary just doing for safe side. we can remove it.